static/app/components/sidebar/sidebarDropdown/switchOrganization.tsx

```diff
@@ -21,67 +23,96 @@ type Props = {
+              // behavior);
+
+              // Stop propagation so that dropdown menu doesn't close here
+              e.stopPropagation();
+            }}
           >
-            <OrganizationList role="list">
-              {sortBy(organizations, ['status.id']).map(organization => {
-                const url = `/organizations/${organization.slug}/`;
-
-                return (
-                  <SidebarMenuItem key={organization.slug} to={url}>
-                    <SidebarOrgSummary organization={organization} />
-                  </SidebarMenuItem>
-                );
-              })}
-            </OrganizationList>
-            {organizations && !!organizations.length && canCreateOrganization && (
-              <Divider css={{marginTop: 0}} />
-            )}
-            {canCreateOrganization && (
-              <SidebarMenuItem
-                data-test-id="sidebar-create-org"
-                to="/organizations/new/"
-                style={{alignItems: 'center'}}
-              >
-                <MenuItemLabelWithIcon>
-                  <StyledIconAdd />
-                  <span>{t('Create a new organization')}</span>
-                </MenuItemLabelWithIcon>
-              </SidebarMenuItem>
-            )}
-          </SwitchOrganizationMenu>
-        )}
-      </Fragment>
-    )}
-  </DeprecatedDropdownMenu>
-);
+            {t('Switch organization')}
+
+            <SubMenuCaret>
+              <IconChevron size="xs" direction="right" />
+            </SubMenuCaret>
+          </SwitchOrganizationMenuActor>
+
+          {isOpen && (
+            <SwitchOrganizationMenu
+              data-test-id="sidebar-switch-org-menu"
+              {...getMenuProps({})}
+            >
+              <OrganizationList role="list">
+                {sortBy(organizations, ['status.id']).map(organization => {
+                  const {slug, links} = organization;
+                  const {organizationUrl} = links;
+
+                  const shouldUseLegacyRoute =
+                    !organizationUrl ||
+                    !organization.features.includes('customer-domains') ||
+                    slug === 'alberto';
+
+                  const menuItemProps: Partial<
+                    React.ComponentProps<typeof SidebarMenuItem>
+                  > = {};
+
+                  if (shouldUseLegacyRoute) {
+                    if (currentOrganization.features.includes('customer-domains')) {
+                      // If the current org is a customer domain, then we need to change the hostname in addition to
+                      // updating the path.
+                      const {sentryUrl} = ConfigStore.get('links');
+                      menuItemProps.href = `${sentryUrl}/organizations/${slug}/`;
+                      menuItemProps.openInNewTab = false;
+                    } else {
+                      menuItemProps.to = `/organizations/${slug}/`;
+                    }
```

Comment:
This feels like a place we could use the frontend path resolution function that was prototyped in one of your other pull requests.

Reply:
@markstory I introduced `useResolveRoute()` which will be a baby step towards org slug-less react routes.